paint/data/stac_client.py

```diff
@@ -0,0 +1,799 @@
+                        )[3][2],
+                        mappings.DATETIME: item.datetime,
+                        "item_id": item.id,
+                    }
+                )
+            # Metadata for properties and deflectometry items are identical.
+            elif (
+                heliostat_collection.id.split("-")[1]
+                == mappings.SAVE_DEFLECTOMETRY.lower()
+                or heliostat_collection.id.split("-")[1].split("_")[1]
+                == mappings.SAVE_PROPERTIES.lower()
+            ):
+                data.append(
+                    {
+                        mappings.HELIOSTAT_ID: child.id.split("-")[0],
+                        mappings.LATITUDE_KEY: item.geometry.get("coordinates")[0],
+                        mappings.LONGITUDE_KEY: item.geometry.get("coordinates")[1],
+                        mappings.ELEVATION: item.geometry.get("coordinates")[2],
+                        mappings.DATETIME: item.datetime,
+                        "item_id": item.id,
+                    }
+                )
+            else:
+                raise ValueError(
+                    f"The collection {heliostat_collection.id.split('-')[0]} is not valid"
+                )
+        pbar.update(1)
+
+    def get_heliostat_metadata(
+        self,
+        heliostats: Union[list[str], None] = None,
+        collections: Union[list[str], None] = None,
+    ) -> None:
+        """
+        Download metadata for desired heliostats.
+
+        Parameters
+        ----------
+        heliostats : list[str], optional
+            Heliostats for which the metadata should be downloaded. If no list is provided, the metadata for all
+            heliostats is downloaded (Default is `None`).
+        collections: list[str], optional
+            Collection for which metadata should be downloaded. If now list is provided, the metadata for all
+            collections is downloaded (Default is `None`).
+        """
+        # Log warning if no collection keys provided.
+        if collections is None:
+            log.warning(
+                "No collections selected - downloading data for all collections!"
+            )
+            collections = [
+                mappings.SAVE_DEFLECTOMETRY.lower(),
+                mappings.SAVE_CALIBRATION.lower(),
+                mappings.SAVE_PROPERTIES.lower(),
+            ]
+        # Check if collection keys provided are acceptable.
+        else:
+            allowed_values = {
+                mappings.SAVE_DEFLECTOMETRY.lower(),
+                mappings.SAVE_CALIBRATION.lower(),
+                mappings.SAVE_PROPERTIES.lower(),
+            }
+            for item in collections:
+                if item not in allowed_values:
+                    raise ValueError(
+                        f"The heliostat collection must be one of: `deflectometry`, `calibration`, "
+                        f"`properties`! The key `{item}` is not accepted!"
+                    )
```

Comment:
This appears three times I think. Maybe we can make a small helper function out of it, like:
```
def check_collection_keys(collections: list[str]) -> None:
    """
    Check whether provided collection keys are valid.

    Parameters
    ----------
    collections : list[str]
        Collection(s) for which (meta)data should be downloaded.

    Raises
    ------
    ValueError
        When a listed collection value is invalid.
    """
    allowed_values = {
                mappings.SAVE_DEFLECTOMETRY.lower(),
                mappings.SAVE_CALIBRATION.lower(),
                mappings.SAVE_PROPERTIES.lower(),
            }
    for item in collections:
        if item not in allowed_values:
            raise ValueError(
                f"The heliostat collection must be one of: `deflectometry`, `calibration`, "
                f"`properties`! The key `{item}` is not accepted!"
            )
```

Reply:
I have included this function for two of the three occurrences - the third occurrence is slightly different since it is used to set the `id_base` used to download data :)